commit c82a8fe6ed07f52bc8969ba849663a8ddfb3cdd6 (HEAD -> master)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2019-08-23T10:57:06+02:00 (7 weeks ago)
    
    TRITON-1875 AdminUI errors in package/image validation for VM provision
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/.gitignore b/.gitignore
index 9638bfdc..82a3476d 100644
--- a/.gitignore
+++ b/.gitignore
@@ -17,3 +17,4 @@ www/libs.js*
 npm-debug.log
 .idea/*
 adminui-pkg-*.gz
+*.vim
diff --git a/www/js/views/provision-vm.js b/www/js/views/provision-vm.js
index 534c276e..9f406e88 100644
--- a/www/js/views/provision-vm.js
+++ b/www/js/views/provision-vm.js
@@ -330,6 +330,7 @@ var View = Backbone.Marionette.Layout.extend({
     },
 
     onSelectImage: function (image) {
+        var self = this;
         if (!image) {
             var imgValue = this.imageInput.$el.val();
             if (!imgValue || imgValue && imgValue.length !== 36) {
@@ -339,6 +340,10 @@ var View = Backbone.Marionette.Layout.extend({
             return;
         }
 
+        if (image) {
+            self.selectedImage = image.attributes;
+        }
+
 /*
  *      Both image and package have an optional brand attribute.
  *      If brand is set in the image, it'll always take precedence.
@@ -470,17 +475,18 @@ var View = Backbone.Marionette.Layout.extend({
     },
 
     extractFormValues: function () {
+        var self = this;
         var formData = this.ui.form.serializeObject();
         var values = {
             image_uuid: formData.image,
             owner_uuid: formData.owner,
             brand: formData.brand
         };
+
         if (formData.delegate_dataset) {
             values.delegate_dataset = true;
         }
 
-
         if (formData.alias && formData.alias.length) {
             values.alias = formData.alias;
         }
@@ -490,7 +496,7 @@ var View = Backbone.Marionette.Layout.extend({
         }
 
         var pkg = this.packages.get(formData['package']);
-        var img = this.images.get(formData['image_uuid']);
+        var img = self.selectedImage;
 
         if (pkg) {
             values['billing_id'] = pkg.get('uuid');
@@ -511,7 +517,7 @@ var View = Backbone.Marionette.Layout.extend({
                     if (disks) {
                         disks[0].image_uuid = values.image_uuid;
                         values['disks'] = disks;
-                    } else {
+                    } else if (img) {
                         values['disks'] = [
                             {'image_uuid': values['image_uuid']},
                             {'size': quotaMib - img.image_size }
